src/program.cpp

```diff
@@ -284,6 +285,10 @@ instruction_ref program::add_outline(const shape& s)
 instruction_ref program::add_parameter(std::string name, shape s)
 {
     assert(get_parameter_shape(name) == shape{});
+    if (name != "scratch")
```

Comment:
Why are we checking for such a parameter name?

Also, we should be compiling with `offload_copy=true`, so this parameter won't get created.

Reply:
The memory_color pass add this parameter, and it is removed later by the dead_code_elimination. Maybe we need to check the flag offload_copy in this pass.

Reply:
Actually, the `preallocate_param` pass removes this parameter regardless of what `offload_copy` is set to

Reply:
Perhaps we need to update the `preallocate_param` pass so it is removing the parameter properly.

Reply:
@scxiao was this resolved?